src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/address_union.rs

```diff
@@ -0,0 +1,449 @@
+use crate::prelude::*;
+
+#[macro_export]
+macro_rules! address_union {
+    (
+        $(
+            #[doc = $expr: expr]
+        )*
+        $union_name: ident,
+        $(
+            $variant_name: ident,
+            $variant_type: ty
+        )+
+    ) => {
+        paste! {
+            $(
+                #[doc = $expr]
+            )*
+            #[derive(
+                Clone,
+                Copy,
+                PartialEq,
+                Eq,
+                Hash,
+                derive_more::Display,
+                derive_more::Debug,
+                uniffi::Enum,
+            )]
+            pub enum $union_name {
+                $(
+                    $variant_name($variant_type),
+                )+
+            }
+
+            #[uniffi::export]
+            pub fn [< new_ $union_name:snake _from_bech32 >](
```

Comment:
`[< new_ $union_name:snake _from_bech32 >]` 

gets substituted for e.g. tagged union `AddressOfAccountOrPersona` with:

`new_address_of_account_or_persona_from_bech32`